packages/backend/src/queryBuilder.ts

```diff
@@ -115,6 +119,15 @@ const renderDateFilterSql = (
             return `(${dimensionSql}) <= ('${dateFormatter(
                 filter.values?.[0],
             )}')`;
+        case FilterOperator.IN_THE_PAST:
```

Comment:
Did you consider doing the "in the past" logic in SQL?

There would be some very big benefits to not pre-calculating the date/time
* Copying the SQL and reusing it would include the logic to do this
* There wouldn't be a delay between compile time and runtime. This could actually make a big difference around time boundaries
* The SQL would show the intent of the query (the user might think "why is this filtering to after a specific date?"
* The generated sql can be cached (this might happen with react query?) in time without invalidating the logic

Reply:
I thought about the functions but not in all the benefits you mention above.
The main reason I went this way instead of the warehouse functions revolve around simplicity.
Pros of this approach:
- bulletproof sql = runs in every warehouse
- easier to test = straightforward logic with few variations
- no extra work each time we add a warehouse connection
- implementation is simple since there is no logic in the FE on what operators are available for each warehouse and apply the correct sql in the BE

Reply:
Related to this, how would we do `is in the past 3 completed months` ?

Reply:
@TuringLovesDeathMetal @owlas was thinking about this code over the weekend. I think we should have a WarehouseBaseClient from where all warehouse client classes extend from and where we build our queries. In this case we would have this generic sql for `renderDateFilterSql` and then each warehouse client would override the function using their specific sql. This way we can have our core generic an unaware of the warehouse being used and it would be simpler to add new warehouse clients. 

Proposal: 
1. leave this logic since this will be the fallback sql since it runs in all warehouses.
2. have ticket to move query builder into WarehouseBaseClient
3. have ticket to add warehouse sql overrides, sql functions, quote char, etc

Reply:
@ZeRego sounds good! 

I'm not the best person to give my input on moving to a `WarehouseBaseClient`. But, what I would say is:
- If the current logic is ready to be shipped now, let's leave it in. Unless it adds a significant amount of tech debt. 
- The more incrementally we can build out this new SQL rendering/compiling approach, the better :) It sounds like the way you're proposing it does that. 

If we move the query builder into WarehouseBaseClient, will this require us to change all fo the warehouse SQL overrides at once?

Reply:
> If we move the query builder into WarehouseBaseClient, will this require us to change all fo the warehouse SQL overrides at once?

No, we can do it incrementally since the WarehouseBaseClient will have a fallback for everything. The fallback sql is basically the code we have now.